src/lib.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright 2018-2019 CrowdStrike, Inc.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+pub mod vcpu;
+
+#[cfg(unix)]
+extern crate kvm_bindings;
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+pub mod x86_64;
```

Comment:
I would suggest to export the structures uniformly on both platforms. Instead of exporting them as mod x86_64 and mod arm, you can export the structures directly.
So basically when you want to use some structures from this crate in an external crate you would write:
`use vmm_vcpu::StandardRegisters` instead of `use vmm_vcpu::x86_64::StandardRegisters`.
 
My thinking here is that x86 and arm are not logical modules, they're just a way for us to separate the code into modules so we don't have a huge file with all defines. What do you think?

Reply:
I am having second doubts about it. Before doing any chance I would try to see how other multi-platform crates are doing this.

Reply:
I went through the same though process having read your comments. On reading your first comment, I was in complete agreement that exporting the structures was cleaner. But having thought it over, I am now leaning toward keeping them as-is (exporting as `mod x86_64` and `mod arm`). The reason being that it forces the code to be explicit that the structures come from specific architectures. This seems particularly important given that the structures can have generic names (`SpecialRegisters`) but represent architecture-dependent structures. Having scanned through various projects on github, I'm not seeing much that _exactly_ fit this use/hierarchy, and so are not directly comparable, but it does seem that exporting the module instead of the structures is more common.

Let me know if you have strong feelings either way, having thought about it over a couple days.